sample/rules.go

```diff
@@ -190,11 +202,21 @@ func ruleMatchesSpanInTrace(trace *types.Trace, rule *config.RulesBasedSamplerRu
 		return true
 	}
 
+	var seen []int
+
 	for _, span := range trace.GetSpans() {
 		ruleMatched := true
-		for _, condition := range rule.Conditions {
-			// whether this condition is matched by this span.
-			value, exists := extractValueFromSpan(trace, span, condition, checkNestedFields)
+		for i, condition := range rule.Conditions {
+			var value interface{}
+			var exists bool
+			// if condition has a root prefix, call extractValueFromSpan and store index in seen
+			if !slices.Contains(seen, i) && strings.Contains(condition.Field, RootPrefix) { // also maybe check Fields
```

Comment:
If a user is designating multiple fields using the `Fields` field (say that five times fast), then should we validate that they either all have root span prefix or none of them do? Should we allow users to mix and match span contexts when specifying multiple fields?

Reply:
I don't think we should forbid it. I could imagine someone saying "if I don't have `user` on the current span, take `root.user`.

Reply:
This has me realizing that `Fields` is sort of an `OR`. Or like a derived column. 🤔🤔🤔

Reply:
added code to support this behavior.